### PR TITLE
feat(header): Implement sticky header :zap:

### PR DESCRIPTION
**Pull request by Athira001**
https://github.com/quintype/ace-planning/issues/185

### CODE REVIEW

File: app/isomorphic/components/atoms/nav-bar-toggle-btn/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import PT from "prop-types";
+
+import "./nav-bar-toggle-btn.m.css";
+
+export const NavBarToggleBtn = ({ onMenuToggle, isMegaMenuOpen }) => {
+  return (
+    <button onClick={onMenuToggle} styleName={`hamburger ${isMegaMenuOpen ? "is-open" : ""}`}>
```
Comment (Athira001) [inline app/isomorphic/components/atoms/nav-bar-toggle-btn/index.js:8]: Check click event on mobile
Rename it to Hambergermneu

File: app/isomorphic/components/header/index.js [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
-      </div>
-    );
-  }
-}
 
-const mapStateToProps = state => ({
-  menu: get(state, ["qt", "data", "navigationMenu"], [])
-});
+const Header = () => {
+  return (
```
Comment (Athira001) [inline app/isomorphic/components/header/index.js:8]: implicit return

File: app/isomorphic/components/header/nav-bar/index.js [hunk 3]
```diff
@@ -1,28 +1,97 @@
...
-              <MenuItem item={item} />
-            </li>
-          );
-        })}
-      </ul>
-      <NavbarSearch />
-    </React.Fragment>
+    <div styleName="main-wrapper" id="sticky-navbar">
+      <nav className="container" styleName="wrapper">
+        {hamburgerMenu.length > 0 && (
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/index.js:53]: use hamburgerMenu.length &&

File: app/isomorphic/components/header/nav-bar/index.js [hunk 4]
```diff
@@ -1,28 +1,97 @@
...
+          <div styleName="dropdown" ref={wrapperRef}>
+            <NavBarToggleBtn onMenuToggle={() => onMenuToggle()} isMegaMenuOpen={isMegaMenuOpen} />
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/index.js:55]: set the state inside  onMenuToggle

File: app/isomorphic/components/header/nav-bar/index.js [hunk 5]
```diff
@@ -1,28 +1,97 @@
...
+            <ul styleName="dropdown-content" style={{ display: isMegaMenuOpen ? "block" : "none" }}>
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/index.js:56]: use variable

File: app/isomorphic/components/header/nav-bar/index.js [hunk 6]
```diff
@@ -1,28 +1,97 @@
...
+                      <MenuItem item={item} showIcon={false} />
+                    </li>
+                  );
+                })}
+            </ul>
+          </div>
+        )}
+
+        <ul styleName="navbar">
+          {menu.length > 0 &&
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/index.js:70]: here as well

File: app/isomorphic/components/header/nav-bar/index.js [hunk 7]
```diff
@@ -1,28 +1,97 @@
...
+                    <li key={item.title} styleName="dropdown">
+                      <MenuItem item={item} showIcon={false} />
+                    </li>
+                  );
+                })}
+            </ul>
+          </div>
+        )}
+
+        <ul styleName="navbar">
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/index.js:69]: put it in a func

File: app/isomorphic/components/header/nav-bar/navbar.m.css [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  margin: 0 12px;
+  display: flex;
+  overflow: auto;
+  font-family: var(--title-font);
+}
+
+.main-wrapper {
+  background-color: var(--brandPrimary);
+  position: relative;
+  box-shadow: -1px 12px 13px -11px rgba(0,0,0,0.42);
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/navbar.m.css:12]: use variable

File: app/isomorphic/components/header/nav-bar/navbar.m.css [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  line-height: 0;
+  flex-shrink: 0;
+}
+
+.dropdown-content {
+  display: none;
+  position: absolute;
+  background-color: var(--white);
+  min-width: 160px;
+  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/navbar.m.css:36]: here as well

File: app/isomorphic/components/header/nav-bar/navbar.m.css [hunk 10]
```diff
@@ -0,0 +1,74 @@
+.navbar {
+  list-style-type: none;
+  margin: 0 12px;
+  display: flex;
+  overflow: auto;
+  font-family: var(--title-font);
+}
+
+.main-wrapper {
+  background-color: var(--brandPrimary);
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/navbar.m.css:10]: keep consistancy

File: app/isomorphic/components/header/nav-bar/navbar.m.css [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+   width: 100%;
+
+  }
+
+  :global .menu-link:last-child {
+    margin-bottom: 0px;
+  }
+
+  :global .menu-link:hover {
+    color: var(--brandPrimaryDark);
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/navbar.m.css:53]: here as well

File: app/isomorphic/components/header/nav-bar/navbar.m.css [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.dropdown-menu {
+  padding: 12px;
+}
+
+.navbar::-webkit-scrollbar {
+  width: 0px;
+  height: 0px;
+  z-index: 9;
```
Comment (Athira001) [inline app/isomorphic/components/header/nav-bar/navbar.m.css:68]: variable

File: app/isomorphic/components/header/top-bar/index.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { AppLogo } from "../app-logo";
+
+import "./top-bar.m.css";
+
+const TopBar = () => {
+  return (
```
Comment (Athira001) [inline app/isomorphic/components/header/top-bar/index.js:7]: remove return

File: app/server/data-loaders/menu-data.js [hunk 14]
```diff
@@ -28,7 +26,9 @@ exports.getNavigationMenuArray = function(menuList, sectionList) {
     .value();
   return {
     footerLinks: menu.filter(item => item["menu-group-slug"] === "footerLinks"),
-    default: menu.filter(item => item["menu-group-slug"] === "default")
+    default: menu.filter(item => item["menu-group-slug"] === "default"),
+    homeMenu: menu.filter(item => item["menu-group-slug"] === "home"),
+    hamburgerMenu: menu.filter(item => item["menu-group-slug"] === "hamburger")
```
Comment (Athira001) [inline app/server/data-loaders/menu-data.js:31]: check for undefined.

File: app/server/handlers/render-layout.js [hunk 15]
```diff
@@ -1,4 +1,9 @@
 /* eslint-disable object-shorthand */
+import React from "react";
+import { ChunkExtractor } from "@loadable/server";
+import path from "path";
+import ReactDOMServer from "react-dom/server";
+import { Provider } from "react-redux";
 import { assetPath, readAsset, getAllChunks } from "@quintype/framework/server/asset-helper";
 import get from "lodash/get";
 import { getChunkName } from "../../isomorphic/pick-component";
```
Comment (Athira001) [inline app/server/handlers/render-layout.js:9]: add space

File: quintype-build.config.js [hunk 16]
```diff
@@ -1 +1,36 @@
...
+        );
+      }
+    });
+
+    return {
+      ...config,
+      plugins: config.plugins.concat([new LoadablePlugin({ writeToDisk: true, filename: path.resolve("stats.json") })]),
+      entry: Object.assign({}, config.entry, {
+        headercss: "./app/isomorphic/components/header/index.js",
+        footercss: "./app/isomorphic/components/layouts/footer/index.js"
```
Comment (Athira001) [inline quintype-build.config.js:26]: no need

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]
Remove DOM manipulations. Block merge.

File: app/isomorphic/components/atoms/nav-bar-toggle-btn/index.js (hunk 1 above)
Comment (cricket-lover) [reply app/isomorphic/components/atoms/nav-bar-toggle-btn/index.js:8]: --done

File: app/isomorphic/components/header/index.js (hunk 2 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/index.js:8]: --done

File: app/isomorphic/components/header/nav-bar/index.js (hunk 3 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/index.js:53]: --done

File: app/isomorphic/components/header/nav-bar/index.js (hunk 5 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/index.js:56]: --done

File: app/isomorphic/components/header/nav-bar/index.js (hunk 7 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/index.js:69]: --done

File: app/isomorphic/components/header/nav-bar/navbar.m.css (hunk 10 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/navbar.m.css:10]: --done

File: app/isomorphic/components/header/nav-bar/navbar.m.css (hunk 8 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/navbar.m.css:12]: --done

File: app/isomorphic/components/header/nav-bar/navbar.m.css (hunk 9 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/navbar.m.css:36]: --done

File: app/isomorphic/components/header/nav-bar/navbar.m.css (hunk 11 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/navbar.m.css:53]: --done

File: app/isomorphic/components/header/nav-bar/navbar.m.css (hunk 12 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/nav-bar/navbar.m.css:68]: --done

File: app/isomorphic/components/header/top-bar/index.js (hunk 13 above)
Comment (cricket-lover) [reply app/isomorphic/components/header/top-bar/index.js:7]: --done

File: app/server/data-loaders/menu-data.js (hunk 14 above)
Comment (cricket-lover) [reply app/server/data-loaders/menu-data.js:31]: undefined check is not necessary as filter don't happen when the array is empty

File: app/assets/stylesheets/base/colors.scss [hunk 17]
```diff
@@ -1,5 +1,13 @@
 :root {
   --white: #ffffff;
-  --rgbBlack: 0, 0, 0;
-  --malibu-gray9: #9c9c9c;
+  --rgb-black: 0, 0, 0;
+  --brand-primary: #2cc7a4;
+  --brand-primary-dark: #108167;
+  --brand-secondary: #1e1e1e;
+  --grey-light: #e8e8e8;
+  --bg-gradient: linear-gradient(to bottom, rgba(76, 76, 76, 0), rgba(59, 59, 59, 0.45) 46%, var(--grey-dark));
```
Comment (Jeevan-Kishore) [inline app/assets/stylesheets/base/colors.scss:8]: eh kya hai?

File: app/assets/stylesheets/base/normalize.scss [hunk 18]
```diff
@@ -0,0 +1,343 @@
+/*! normalize.css v8.0.0 | MIT License | github.com/necolas/normalize.css */
```
Comment (Jeevan-Kishore) [inline app/assets/stylesheets/base/normalize.scss:1]: Is this needed @nandap4790 ?

File: app/assets/stylesheets/base/typography.scss [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+  --semi-bold: 600;
+  --bold: 700;
+}
+
+a {
+  text-decoration: none;
+  color: inherit;
+}
+
+html {
```
Comment (Jeevan-Kishore) [inline app/assets/stylesheets/base/typography.scss:24]: Most styles in this file are not needed

File: app/assets/stylesheets/base/z-index.scss [hunk 20]
```diff
@@ -1,4 +1,9 @@
 :root {
-  --z-level-1: 100;
+  --zindex-1: -1;
```
Comment (Jeevan-Kishore) [inline app/assets/stylesheets/base/z-index.scss:2]: rename

File: app/isomorphic/components/atoms/hamburger-menu/index.js [hunk 21]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import PT from "prop-types";
+
+import "./hamburger-menu.m.css";
+
+export const HamburgerMenu = ({ onMenuToggle, isMegaMenuOpen }) => {
+  return (
+    <button onClick={onMenuToggle} styleName={`hamburger ${isMegaMenuOpen ? "is-open" : ""}`}>
+      <span styleName="line" />
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/atoms/hamburger-menu/index.js:9]: loop it

File: app/isomorphic/components/header/index.js [hunk 22]
```diff
@@ -1,24 +1,13 @@
...
-      </div>
-    );
-  }
-}
 
-const mapStateToProps = state => ({
-  menu: get(state, ["qt", "data", "navigationMenu"], [])
-});
+const Header = () => (
+  <React.Fragment>
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/index.js:8]: import fragment from react

File: app/isomorphic/components/header/menu-item/index.js [hunk 23]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import PT from "prop-types";
+import { Link } from "@quintype/components";
+import assetify from "@quintype/framework/assetify";
+
+import ArrowDownIcon from "../../atoms/icons/arrow-down.svg";
+
+import "./menu-item.m.css";
+
+function MenuItem({ item, showIcon = true }) {
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/menu-item/index.js:10]: use arrow

File: app/isomorphic/components/header/menu-item/index.js [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+  return item.isExternalLink ? (
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/menu-item/index.js:11]: change to `if`

File: app/isomorphic/components/header/menu-item/index.js [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+    <a styleName="menu-link" className="menu-link" target="_blank" rel="noopener noreferrer" href={item.url || "/"}>
+      {item.title}{" "}
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/menu-item/index.js:13]: remove `{" "}`

File: app/isomorphic/components/header/menu-item/index.js [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+      {item.children.length >= 1 && showIcon && (
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/menu-item/index.js:14]:
```suggestion
      {item.children.length && showIcon && (
```

Comment (Jeevan-Kishore) [reply app/isomorphic/components/header/menu-item/index.js:14]: Should be a function / if

File: app/isomorphic/components/header/menu-item/index.js [hunk 27]
```diff
@@ -0,0 +1,31 @@
...
+        <img width="16px" height="16px" src={`${assetify(ArrowDownIcon)}`} alt="down-arrow" />
+      )}
+    </a>
+  ) : (
+    <Link styleName="menu-link" className="menu-link" href={item.url || "/"}>
+      {item.title}{" "}
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/menu-item/index.js:20]:
```suggestion
      {item.title}
```

File: app/isomorphic/components/header/menu-item/index.js [hunk 28]
```diff
@@ -0,0 +1,31 @@
...
+      {item.children.length >= 1 && showIcon && <img src={`${assetify(ArrowDownIcon)}`} alt="down-arrow" />}
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/menu-item/index.js:21]:
```suggestion
      {item.children.length && showIcon && <img src={`${assetify(ArrowDownIcon)}`} alt="down-arrow" />}
```

File: app/isomorphic/components/header/nav-bar/index.js [hunk 29]
```diff
@@ -1,28 +1,72 @@
...
+    <ul styleName="navbar">
+      {menu.length &&
+        menu.map(item => {
           return (
-            <li key={`${item.id}${index}`} styleName="menu-item desktop-view">
+            <li key={item.title} styleName="dropdown">
               <MenuItem item={item} />
+              {item.children && item.children.length > 0 && (
+                <div styleName="dropdown-content">
+                  <ul style={{ position: "relative" }}>
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/nav-bar/index.js:21]: inline styles?

File: app/isomorphic/components/header/nav-bar/index.js [hunk 30]
```diff
@@ -1,28 +1,72 @@
...
-      <ul styleName="navbar">
-        {get(menu, ["default"], []).map((item, index) => {
+    <ul styleName="navbar">
+      {menu.length &&
+        menu.map(item => {
           return (
-            <li key={`${item.id}${index}`} styleName="menu-item desktop-view">
+            <li key={item.title} styleName="dropdown">
               <MenuItem item={item} />
+              {item.children && item.children.length > 0 && (
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/nav-bar/index.js:19]:
```suggestion
              {item.children && (
```

File: app/isomorphic/components/header/nav-bar/index.js [hunk 31]
```diff
@@ -1,28 +1,72 @@
...
-      </ul>
-      <NavbarSearch />
-    </React.Fragment>
+    </ul>
+  );
+};
+
+const NavBar = () => {
+  const [isMegaMenuOpen, setIsMegaMenuOpen] = useState(false);
+  const wrapperRef = useRef(null);
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/nav-bar/index.js:41]:
```suggestion

```

File: app/isomorphic/components/header/nav-bar/index.js [hunk 32]
```diff
@@ -1,28 +1,72 @@
...
+  return (
+    <div styleName="main-wrapper" id="sticky-navbar">
+      <nav className="container" styleName="wrapper">
+        {hamburgerMenu.length && (
+          <div styleName="dropdown" ref={wrapperRef}>
+            <HamburgerMenu onMenuToggle={() => setIsMegaMenuOpen(!isMegaMenuOpen)} isMegaMenuOpen={isMegaMenuOpen} />
+            <div styleName="overlay" onClick={()=> setIsMegaMenuOpen(false)}></div>
+            <ul styleName="dropdown-content" style={{ display: displayStyle }}>
+              {isMegaMenuOpen &&
+                hamburgerMenu.map(item => {
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/nav-bar/index.js:56]: func

File: app/isomorphic/components/header/nav-bar/index.js [hunk 33]
```diff
@@ -1,28 +1,72 @@
...
+                    <li key={item.title} styleName="dropdown">
+                      <MenuItem item={item} showIcon={false} />
+                    </li>
+                  );
+                })}
+            </ul>
+          </div>
+        )}
+        <div>{getNavbar(menu)}</div>
+        <div> user</div>
```
Comment (Jeevan-Kishore) [inline app/isomorphic/components/header/nav-bar/index.js:67]: ?

File: app/server/data-loaders/menu-data.js [hunk 34]
```diff
@@ -43,9 +43,7 @@ function findCompleteUrl(menutItem, sectionList) {
     const parentSectionObj = _.find(sectionList, function(item) {
       return sectionObject["parent-id"] === item.id;
     });
-    return parentSectionObj
-      ? "/" + parentSectionObj.slug + "/" + sectionObject.slug
-      : "/#";
+    return parentSectionObj ? "/" + parentSectionObj.slug + "/" + sectionObject.slug : "/#";
```
Comment (Jeevan-Kishore) [inline app/server/data-loaders/menu-data.js:46]: use template strings

File: app/server/data-loaders/menu-data.js (hunk 14 above)
Comment (Jeevan-Kishore) [reply app/server/data-loaders/menu-data.js:31]: +1

File: app/server/handlers/render-layout.js [hunk 35]
```diff
@@ -8,10 +13,19 @@ import { Footer } from "../../isomorphic/components/layouts/footer";
...
 import { BreakingNewsView } from "../../isomorphic/components/breaking-news-view";
 import serialize from "serialize-javascript";
+
+const statsFile = path.resolve("stats.json");
 const cssContent = assetPath("app.css") ? readAsset("app.css") : "";
 const fontJsContent = assetPath("font.js") ? readAsset("font.js") : "";
 const allChunks = getAllChunks("list", "story");
 
+function renderLoadableReduxComponent(Component, store, extractor, props) {
+  const comp = extractor.collectChunks(React.createElement(Provider, { store }, React.createElement(Component, props)));
```
Comment (Jeevan-Kishore) [inline app/server/handlers/render-layout.js:23]:
```suggestion
  const children = React.createElement(Component, props);
  const comp = extractor.collectChunks(React.createElement(Provider, { store }, children));
```

File: app/server/handlers/render-layout.js [hunk 36]
```diff
@@ -8,10 +13,19 @@ import { Footer } from "../../isomorphic/components/layouts/footer";
...
+  const string = ReactDOMServer.renderToString(comp);
```
Comment (Jeevan-Kishore) [inline app/server/handlers/render-layout.js:24]:
```suggestion
  const renderString = ReactDOMServer.renderToString(comp);
```

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/base/colors.scss (hunk 17 above)
Comment (Athira001) [reply app/assets/stylesheets/base/colors.scss:8]: Removed it. not using

File: app/assets/stylesheets/base/z-index.scss (hunk 20 above)
Comment (Athira001) [reply app/assets/stylesheets/base/z-index.scss:2]: done

File: app/isomorphic/components/atoms/hamburger-menu/index.js (hunk 21 above)
Comment (Athira001) [reply app/isomorphic/components/atoms/hamburger-menu/index.js:9]: done

File: app/isomorphic/components/header/nav-bar/index.js (hunk 33 above)
Comment (Athira001) [reply app/isomorphic/components/header/nav-bar/index.js:67]: will implement later

File: app/server/data-loaders/menu-data.js (hunk 34 above)
Comment (Athira001) [reply app/server/data-loaders/menu-data.js:46]: done

File: app/server/handlers/render-layout.js (hunk 35 above)
Comment (Athira001) [reply app/server/handlers/render-layout.js:23]: done

File: app/server/handlers/render-layout.js (hunk 36 above)
Comment (Athira001) [reply app/server/handlers/render-layout.js:24]: done

File: app/assets/stylesheets/base/typography.scss [hunk 37]
```diff
@@ -0,0 +1,58 @@
...
+  margin: 0;
+  padding: 0;
+  font-family: var(--content-font);
+}
+
+p,
+li {
+  font-size: var(--base);
+}
+small {
```
Comment (Harshith46) [inline app/assets/stylesheets/base/typography.scss:55]: are we using a small tag?

File: app/isomorphic/components/atoms/hamburger-menu/hamburger-menu.m.css [hunk 38]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.hamburger:focus {
+  outline: none;
+}
+
+.line {
+  background-color: var(--brand-secondary);
+  display: block;
+  margin-bottom: 3px;
```
Comment (Harshith46) [inline app/isomorphic/components/atoms/hamburger-menu/hamburger-menu.m.css:17]: even number

File: app/isomorphic/components/atoms/hamburger-menu/index.js [hunk 39]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import PT from "prop-types";
+
+import "./hamburger-menu.m.css";
+
+export const HamburgerMenu = ({ onMenuToggle, isMegaMenuOpen }) => {
+  return (
```
Comment (Harshith46) [inline app/isomorphic/components/atoms/hamburger-menu/index.js:7]: no need for a return

File: app/isomorphic/components/breaking-news-view.js [hunk 40]
```diff
@@ -49,7 +49,7 @@ export const BreakingNewsViewBase = ({ breakingNews = [], breakingNewsConfig = {
   }
 
   return (
-    <div styleName="base">
+    <div styleName="base" className="container">
```
Comment (Harshith46) [inline app/isomorphic/components/breaking-news-view.js:52]: can we keep classname specific to this component, the container makes a generic name

File: app/isomorphic/components/header/app-logo/index.js [hunk 41]
```diff
@@ -2,15 +2,22 @@ import React from "react";
...
 import assetify from "@quintype/framework/assetify";
 
-import "./styles.m.css";
+import "./app-logo.m.css";
 
-import logo from "./publisher-logo.png";
+import logo from "../../../../assets/images/malibu-logo.png";
 
 function AppLogo() {
   return (
```
Comment (Harshith46) [inline app/isomorphic/components/header/app-logo/index.js:10]: no need for return

File: app/isomorphic/components/header/menu-item/index.js [hunk 42]
```diff
@@ -0,0 +1,34 @@
...
+  if (item.isExternalLink) {
+    return (
+      <a styleName="menu-link" className="menu-link" target="_blank" rel="noopener noreferrer" href={item.url || "/"}>
+        {item.title}
+        {item.children.length > 0 && showIcon && (
+          <img width="16px" height="16px" src={`${assetify(ArrowDownIcon)}`} alt="down-arrow" />
+        )}
+      </a>
+    );
+  } else {
```
Comment (Harshith46) [inline app/isomorphic/components/header/menu-item/index.js:19]: no need of else block, you can directly do return {
}

File: app/isomorphic/components/header/menu-item/index.js [hunk 43]
```diff
@@ -0,0 +1,34 @@
...
+      <Link styleName="menu-link" className="menu-link" href={item.url || "/"}>
+        {item.title}
+        {item.children.length > 0 && showIcon && <img src={`${assetify(ArrowDownIcon)}`} alt="down-arrow" />}
+      </Link>
+    );
+  }
+};
+
+MenuItem.propTypes = {
+  item: PT.object,
```
Comment (Harshith46) [inline app/isomorphic/components/header/menu-item/index.js:30]: import only what is required

File: app/isomorphic/components/header/nav-bar/index.js [hunk 44]
```diff
@@ -1,28 +1,76 @@
...
-import { AppLogo } from "../app-logo";
+import { MenuItem } from "../menu-item";
+import HamburgerMenu from "../../atoms/hamburger-menu";
 
-import "./styles.m.css";
+import "./navbar.m.css";
 
-const NavBar = ({ menu }) => {
+const getNavbarMenu = menu => {
   return (
```
Comment (Harshith46) [inline app/isomorphic/components/header/nav-bar/index.js:12]: remove return

File: app/isomorphic/components/header/top-bar/top-bar.m.css [hunk 45]
```diff
@@ -0,0 +1,11 @@
+.main-wrapper {
+  position: relative;
+  z-index: var(--zlevel0);
+  padding: 9px 0;
```
Comment (Harshith46) [inline app/isomorphic/components/header/top-bar/top-bar.m.css:4]: even value

File: app/isomorphic/components/pages/author-page/index.js [hunk 46]
```diff
@@ -9,10 +9,10 @@ const AuthorPage = props => {
     return <h1>No author found</h1>;
   }
   return (
-    <React.Fragment>
+    <div className="container">
```
Comment (Harshith46) [inline app/isomorphic/components/pages/author-page/index.js:12]: classname let it be specific to this component

File: app/isomorphic/components/pages/home.js [hunk 47]
```diff
@@ -6,7 +6,7 @@ import img from "../../../assets/images/pw_maze_white.png";
 import { getCollectionTemplate } from "../get-collection-template";
 
 export const HomePage = props => (
-  <div>
+  <div className="container">
```
Comment (Harshith46) [inline app/isomorphic/components/pages/home.js:9]: same here or do you want styles of container classname ?

File: app/isomorphic/components/pages/search.js [hunk 48]
```diff
@@ -1,14 +1,19 @@
...
+  <div className="container">
     <h1>
       Search - {props.data.query} ({props.data.total} results)
     </h1>
     <StoryGrid stories={props.data.stories} />
   </div>
 );
 
+SearchPage.propTypes = {
+  data: PropTypes.object
```
Comment (Harshith46) [inline app/isomorphic/components/pages/search.js:16]: import only required items

**Review by Harshith46 — CHANGES_REQUESTED** [review]

**Comment by Athira001** [comment]
@Harshith46  container class is defined globally for setting the width of the container across the app.

**Review by Harshith46 — APPROVED** [review]

**Review by Jeevan-Kishore — APPROVED** [review]
Approving, on comments resolved

**Comment by Athira001** [comment]
Support Loadable Component
Install 
`npm install @loadable/server && npm install --save-dev @loadable/babel-plugin @loadable/webpack-plugin`
Overwrite the file `quintype-build-config.js` and modify the webpack config  to support @loadable/webpack-plugin and @loadable/babel-plugin
```
const { produce } = require("immer");
const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
const LoadablePlugin = require("@loadable/webpack-plugin");
const path = require("path");
module.exports = {
  modifyWebpackConfig: function({ defaultConfig }) {
    const config = produce(defaultConfig, function(draft) {
      draft.node = { Buffer: false };
      draft.entry.font = "./app/client/font.js";
      if (process.env.ANALYZE_STATS === "true") {
        draft.plugins.push(
          new BundleAnalyzerPlugin({
            generateStatsFile: true,
            analyzerMode: "static",
            statsFilename: "prod-stats.json"
          })
        );
      }
    });

    return {
      ...config,
      plugins: config.plugins.concat([new LoadablePlugin({ writeToDisk: true, filename: path.resolve("stats.json") })]),
      entry: Object.assign({}, config.entry, {
        topbarCriticalCss: "./app/isomorphic/components/header", //When running your build, notice @loadable/webpack-plugin generates a file called stats.json, which contains information about all your entries and chuncks from webpack.
        navbarCriticalCss: "./app/isomorphic/components/header/nav-bar"
      })
    };
  },
  modifyBabelConfig: function({ defaultConfig, babelTarget, env }) {
    const config = produce(defaultConfig, draft => {
      draft.plugins.push(["@loadable/babel-plugin"]);
    });
    return { ...config };
  }
};

```

**Pull request merged**